lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_caches.jl

```diff
@@ -1072,20 +974,9 @@ function alg_cache(
         ::Type{uBottomEltypeNoUnits}, ::Type{tTypeNoUnits}, uprev, uprev2, f, t,
         dt, reltol, p, calck,
         ::Val{true}) where {uEltypeNoUnits, uBottomEltypeNoUnits, tTypeNoUnits}
-    dense1 = zero(rate_prototype)
-    dense2 = zero(rate_prototype)
-    dense3 = zero(rate_prototype)
-    du = zero(rate_prototype)
-    du1 = zero(rate_prototype)
-    du2 = zero(rate_prototype)
-    k1 = zero(rate_prototype)
-    k2 = zero(rate_prototype)
-    k3 = zero(rate_prototype)
-    k4 = zero(rate_prototype)
-    k5 = zero(rate_prototype)
-    k6 = zero(rate_prototype)
-    k7 = zero(rate_prototype)
-    k8 = zero(rate_prototype)
+    dense = zeros(eltype(rate_prototype), 3)
```

Comment:
```suggestion
    dense = [zero(rate_protoype) for i in 1:3]
```

otherwise it's not actually a zero vector.